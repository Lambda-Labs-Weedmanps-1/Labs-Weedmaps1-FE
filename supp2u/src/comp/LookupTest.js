import React, { Component } from 'react';

import axios from 'axios';

class LookupTest extends Component {
    constructor() {
        super()

        this.state = {
            address: "",
            lat: "32342",
            lon: "23432",
        }

    }


    handleAddress = e => {
        this.setState({
          ...this.state,
          
          address: e.target.value
          
        });
      };

      getCoordinates = addy => {

      }



      getLL = addy => {
        addy.preventDefault();

        // axios.get(`https://maps.googleapis.com/maps/api/geocode/json?address=1600+Amphitheatre+Parkway,+Mountain+View,+CA&key=YOUR_API_KEY`)
    
        axios.get(`https://maps.googleapis.com/maps/api/geocode/json?address=${this.state.address}=AIzaSyDtc3pDlAa_YG-MOK-irmjZAM_fFJbpEgg`)
        .then (res => {
            console.log("res", res);
            console.log("res data", res.data)
        })
        
    
    }



    render() {
        return (
            <div>
                <h1>LookupTest</h1>
                <p>Enter an address, to search for geocoordinates</p>
                
                <form onSubmit= {this.getLL}>
                    <input
                        type="text"
                        name="address"
                        placeholder="address"
                        onChange={this.handleAddress}
                        value={this.state.address}
                    />
                </form>

                <br></br><br></br>

                <p>geocoordinates:</p>
                <p>Lat: {this.state.lat}</p>
                <p>Lon: {this.state.lon}</p>

                <br></br><br></br>

            </div>
        )
    }


}

export default LookupTest;