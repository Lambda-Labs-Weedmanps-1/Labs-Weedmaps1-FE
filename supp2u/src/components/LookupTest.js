import React, { Component } from 'react';

import axios from 'axios';



class LookupTest extends Component {
    constructor() {
        super()

        this.state = {
            address: "",
            lat: "32342",
            lon: "23432",
        }

    }


    handleAddress = e => {
        this.setState({
          ...this.state,
          
          address: e.target.value
          
        });
      };

      getCoordinates = addy => {

      }



      getLL = addy => {
        addy.preventDefault();

        // This is currently just a test address and not taking in the actual state address
        // axios.get(`https://maps.googleapis.com/maps/api/geocode/json?address=1600+Amphitheatre+Parkway,+Mountain+View,+CA&key=XXXXXXXXXXXXXXXXXXX`)
    
        axios.get(`https://maps.googleapis.com/maps/api/geocode/json?address=1600+Amphitheatre+Parkway,+Mountain+View,+CA&key=${process.env.REACT_APP_GCOORDINATES}`)

        .then (res => {
            console.log("res", res);
            console.log("res data", res.data)
        })
        
    
    }



    render() {
        return (
            <div>
                <h1>LookupTest</h1>
                <p>Enter an address, to search for geocoordinates</p>
                
                <form onSubmit= {this.getLL}>
                    <input
                        type="text"
                        name="address"
                        placeholder="address"
                        onChange={this.handleAddress}
                        value={this.state.address}
                    />
                </form>

                <br></br><br></br>

                <p>geocoordinates:</p>
                <p>Lat: {this.state.lat}</p>
                <p>Lon: {this.state.lon}</p>

                <br></br><br></br>

                <p>{process.env.TESTDATA}</p>

            </div>
        )
    }


}

export default LookupTest;