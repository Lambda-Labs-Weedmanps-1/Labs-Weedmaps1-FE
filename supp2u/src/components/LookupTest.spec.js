import React from 'react';
import { mount, configure, shallow, enzyme } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import LookupTest from './LookupTest'

configure({ adapter: new Adapter() });

describe('AddCustomer Renders', () => {
    // If Map component render is ever non functional, this should fail
    it('should render correctly when component called', () => {
      const Customer = shallow(<LookupTest />);

      expect(Customer).toMatchSnapshot();
    });
});

describe("testing updating form", () => {
    it('should submit form', () => {
        // const submit = jest.fn();
        let wrapper = shallow(<LookupTest />)
        const preventDefault = jest.fn();
        wrapper.find('form').simulate('submit', {preventDefault});
        expect(preventDefault).toHaveBeenCalled()
    })
})