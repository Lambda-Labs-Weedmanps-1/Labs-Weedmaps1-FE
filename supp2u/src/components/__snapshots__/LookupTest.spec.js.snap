// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCustomer Renders should render correctly when component called 1`] = `
<div>
  <h1>
    LookupTest
  </h1>
  <p>
    Enter an address, to search for geocoordinates
  </p>
  <form
    onSubmit={[Function]}
  >
    <input
      name="address"
      onChange={[Function]}
      placeholder="address"
      type="text"
      value=""
    />
  </form>
  <br />
  <br />
  <p>
    geocoordinates:
  </p>
  <p>
    Lat: 
    32342
  </p>
  <p>
    Lon: 
    23432
  </p>
  <br />
  <br />
  <p />
</div>
`;
